test/org/javarosa/TestHelper.java

```diff
@@ -0,0 +1,93 @@
+package org.javarosa;
+
+import static org.javarosa.core.model.instance.TreeReference.CONTEXT_ABSOLUTE;
+import static org.javarosa.core.model.instance.TreeReference.INDEX_UNBOUND;
+import static org.javarosa.core.model.instance.TreeReference.REF_ABSOLUTE;
+
+import java.util.Arrays;
+import java.util.List;
+import org.javarosa.core.model.FormDef;
+import org.javarosa.core.model.FormIndex;
+import org.javarosa.core.model.SelectChoice;
+import org.javarosa.core.model.instance.InstanceInitializationFactory;
+import org.javarosa.core.model.instance.TreeReference;
+import org.javarosa.form.api.FormEntryPrompt;
+
+public class TestHelper {
+    /**
+     * Builds an absolute {@link TreeReference} from a string path description
+     * like <code>/some/path</code>.
+     * <p>
+     * Can be used in combination of other methods that take {@link TreeReference},
+     * like {@link #getFormIndex(FormDef, TreeReference)} or {@link #getSelectChoices(FormDef, String)}
+     * to make them easier to use. Example:
+     *
+     * <code>FormIndex formIndex = getFormIndex(formDef, absoluteRef("/data/field"));</code>
+     */
+    public static TreeReference absoluteRef(String path) {
+        TreeReference tr = new TreeReference();
+        tr.setRefLevel(REF_ABSOLUTE);
+        tr.setContext(CONTEXT_ABSOLUTE);
+        tr.setInstanceName(null);
+        Arrays.stream(path.split("/"))
+            .filter(s -> !s.isEmpty())
+            .forEach(s -> tr.add(s, INDEX_UNBOUND));
+        return tr;
+    }
+
+    /**
+     * Returns a {@link FormIndex} corresponding to the given string reference.
+     * <p>
+     * The absolute {@link TreeReference} of the question is built with {@link #absoluteRef(String)}.
+     */
+    public static FormIndex getFormIndex(FormDef formDef, String ref) {
+        return getFormIndex(formDef, absoluteRef(ref));
+    }
+
+    /**
+     * Returns a {@link FormIndex} corresponding to the given {@link TreeReference}.
+     */
+    public static FormIndex getFormIndex(FormDef formDef, TreeReference ref) {
```

Comment:
This would need to be recursive to work for a depth more than 1 (nodes in groups or repeats). I'm starting to prefer what I did in https://github.com/opendatakit/javarosa/commit/ee5527fd3c2d49549e288de6beaab5df48068be8 because it guarantees it gets something that's actually in the form with all the correct multiplicities, context, etc set. I also like that it's not in a grab-bag of static methods. Do you prefer keeping all methods most helpful for testing together?

Reply:
Yeah, this only makes sense for the tests I've used it with, which doesn't include groups. 

I have many doubts about all the methods in this class because I haven't worked with JR enough to understand what's useful and what's not. For example, in your ee5527f you return a `TreeElement` instead of a `FormIndex`. Is that more generally useful or just for the tests you're writing? `¯\_(ツ)_/¯`

If it turns out that the implementation in ee5527f is more useful, I'd suggest we improve this method with it and you use it in your PR.

In my approach, this would be the first step to generalize a BDD harness for tests: get every test utility method into the same bag and start figuring out what structures emerge. This is not something that we can do in one PR and we will have lots of duplicated ugly code, but that'd be by design.

Another approach would be to forget about that and start top-down implementing everything required to write this:
```
Context(form("the_form.xml"))
  .given(answer("/the/ref/to/question1", "some value"))
  .given(answer("/the/ref/to/question2", "some other value"))

  .when(answer("/the/ref/to/question3", "yet another value"))

  .then(question("/the/ref/to/question4"), contains(choices("a", "b", "c")));
```
We can do that too and take what we do from all the different tests we already have. I can prepare a scaffolding for that but I'd need help to write the implementations